refinery/file_store/models.py

```diff
@@ -64,8 +61,11 @@ def file_path(instance, filename):
     # provides 256 * 256 = 65536 of possible directory combinations
     dir1 = "{:0>2x}".format(hashcode & mask)
     dir2 = "{:0>2x}".format((hashcode >> 8) & mask)
-    # Galaxy doesn't process names with parentheses in them
-    filename = re.sub('[()]', '_', filename)
+    # make file name compliant with POSIX "fully portable filenames"
+    # max length: 255, allowed character set: [A-Za-z0-9._-]
+    # remove leading '-' characters to make file management easier
+    filename = re.sub('[^A-Za-z0-9._-]', '_',
+                      string.lstrip(filename, '-'))[-255:]
```

Comment:
Is the idea that the last part of the filename has the extension, and you definitely want to preserve that? Hopefully we never get files this long, but for people, the start of the name is also significant... it'd be more complicated, and probably not worth it, but I could imagine removing the part of the name just before the extension.

Reply:
I think most file systems in use today have a limit of 255 on file name length which means there should really be no files that are above that. So this is more of a last line of defense for which simple handling is sufficient.